npm/translatePolicy.go

```diff
@@ -19,6 +20,19 @@ type portsInfo struct {
 	port     string
 }
 
+// getPortRange returns a specific port (only portRule.Port exist) or port ranges (when EndPort exists as well) based on portRule
+func getPortRange(portRule networkingv1.NetworkPolicyPort) (string, bool) {
+	if portRule.Port == nil {
+		return "", false
+	}
+
+	portRange := portRule.Port.String()
+	if portRule.EndPort != nil {
```

Comment:
I know API would have done validations, but can we do some validations too to make sure endport is always greater than start port.

Reply:
This is a valid point.  I also considered this and decided not to check it in the NPM code since API server validates the input and end-port is a spec in network policy since I think this is a string contract.
In addition, even though we check the validation in NPM and "end port" is less than "start port" (while it does not happen without our code bug), what is the right action in NPM? I think there are no right actions for this.   

Please let me know you have other opinions (or string opinion to add the logic).

Btw, I am curious how iptables react when I insert "[endport]:[start port]" order in iptables. I will check it.

Reply:
Any update on the check you mentioned above? you are right, API server will maintain that sanity checks and we can trust it.